spec/lib/artifact-search.spec.js

```diff
@@ -0,0 +1,135 @@
+                search: [
+                    "example=123",
+                    "example=456"
+                ]
+            };
+
+            return instance.search(searchParameters).then((urls) => {
+                expect(urls).toEqual([
+                    "/shelf-js-test/an-artifact/2016-11-23T16:03:05.239Z",
+                    "/shelf-js-test/an-artifact/2016-11-23T16:01:29.716Z"
+                ]);
+            });
+        });
+        it("catches errors from requestPromise", () => {
+            requestPromiseMock.post.andReturn(bluebird.reject({
+                code: "exampleTimeoutError"
+            }));
+
+            return instance.search().then(() => {
+                expect(responseHandler.handleErrorResponse).toHaveBeenCalledWith({
+                    code: "exampleTimeoutError"
+                });
+            });
+        });
+        describe(".validateSearchItem()", () => {
+            it("rejects if a search value doesn't provide the section when using the search string form =", () => {
+                /* eslint-disable no-useless-escape */
+                searchParameters = {
+                    search: "Invalid\\=Query"
+                };
+
+                /* eslint-enable no-useless-escape */
+
+                return instance.search(searchParameters).then(jasmine.fail, () => {
+                    expect(bluebird.reject).toHaveBeenCalled();
+                });
+            });
+            it("resolves a valid query", () => {
+                searchParameters = {
+                    search: "Valid=Query"
+                };
+
+                return instance.search(searchParameters).then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("resolves a valid query with '=*'", () => {
+                searchParameters = {
+                    search: "Valid=*Query"
+                };
+
+                return instance.search(searchParameters).then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("resolves a valid query with ~=", () => {
+                searchParameters = {
+                    search: "Valid~=Query"
+                };
+
+                return instance.search(searchParameters).then(() => {
+                    expect(bluebird.resolve).toHaveBeenCalled();
+                });
+            });
+            it("rejects if the query isn't a string", () => {
+                searchParameters = {
+                    search: {
+                        query: "This won't work"
+                    }
+                };
+
+                return instance.search(searchParameters).then(jasmine.fail, () => {
+                    expect(bluebird.reject).toHaveBeenCalled();
+                });
+            });
+        });
+        describe(".validateSearch()", () => {
+            it("resolves the promise if the query list is falsy", () => {
```

Comment:
The queryList being undefined is the most likely scenario, but I could see some weird cases where it is null or false or an empty object/array